client/src/app/tabular/tabular.component.ts

```diff
@@ -0,0 +1,77 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { UserModel, columnName} from '../user.model';
+import { UserService } from '../user.service';
+
+@Component({
+  selector: 'app-tabular',
+  templateUrl: './tabular.component.html',
+  styleUrls: ['./tabular.component.css']
+})
+export class TabularComponent implements OnInit {
+  prevArray:string[] = [];
+  @Input ('userdata') userData:UserModel[];
+  saveEnable = false;
+  header = Object.values(columnName);
+
+  constructor(private userService: UserService) { }
+
+  ngOnInit(): void {
+  }
+
+  onUpdatedData(){
+    this.userService.getUsers().subscribe(newData=>{
+      this.userData = newData;
+    });
+  }
+
+  onEdit(event:any){
+    event.path[0].value = "Save";
+    event.path[1].childNodes[1].value = "Cancel";
+    for(let j=0;j<event.path[2].cells.length-4;j++){
+      event.path[2].cells[j].childNodes[0].disabled = false;
+      this.prevArray.push(event.path[2].cells[j].childNodes[0].value);
+    }
+    this.saveEnable = true
+  }
+  
+  onSave(event:any,data:UserModel){
+    this.saveEnable = false;
+    let editData: {[key: string]: string|number}={};
+    event.path[0].value = "Edit";
```

Comment:
add null checks here..
Also, Please explain what we are doing here, Why do we need such manipulations,
Can't we handle them via ngif directive

Reply:
Made changes in 3 files
1) user.model.ts- added a property 'saveEnable'
2) tabular.component.ts- removed all the lengthy 'saveEnabled' statments and instead used saveEnable as a flag to enable those functionality
3) tabular.component.html - used 'saveEnable' flag to avoid specifying save, delete again and again as discussed in the meet